src/Libraries/PythonNodeModelsWpf/ScriptEditorWindow.xaml.cs

```diff
@@ -117,22 +122,37 @@ private void OnTextAreaTextEntered(object sender, TextCompositionEventArgs e)
 
         private void OnAcceptClicked(object sender, RoutedEventArgs e)
         {
-            DialogResult = true;
+            UpdateScript(editText.Text);
+            this.Close();
         }
 
         private void OnCancelClicked(object sender, RoutedEventArgs e)
         {
-            DialogResult = false;
+            if (nodeWasModified)
+            {
+                UpdateScript(originalScript);
+            }
+            
+            this.Close();
         }
 
-        private void OnScriptEditWindowClosed(object sender, EventArgs e)
+        private void UpdateScript(string scriptText)
         {
-            if (DialogResult.HasValue && (DialogResult.Value))
-            {
-                var command = new DynamoModel.UpdateModelValueCommand(
-                    System.Guid.Empty, boundNodeId, propertyName, editText.Text);
+            var command = new DynamoModel.UpdateModelValueCommand(
+                System.Guid.Empty, boundNodeId, propertyName, scriptText);
 
-                dynamoViewModel.ExecuteCommand(command);
+            dynamoViewModel.ExecuteCommand(command);
+            this.Focus();
+            nodeWasModified = true;
+            nodeModel.OnNodeModified();
+        }
+
+        private void OnTestClicked(object sender, RoutedEventArgs e)
+        {
+            UpdateScript(editText.Text);
+            if (dynamoViewModel.HomeSpace.RunSettings.RunType != RunType.Automatic)
```

Comment:
So the `Run` button in the editor will have no effect unless the run mode in Dynamo is set to `Automatic` and will have no effect if the mode is set to manual Run? Not sure if some users may expect this Run button to override the Dynamo mode setting and still execute the graph?

Reply:
we would expect it to do that :)

Reply:
Run always triggers a re-execution of the changes to the Python node. In automatic mode, it merely updates the script and marks the node as modified; automatic mode takes care of the rest. In manual or periodic mode, an execution is explicitly triggered.

Reply:
Ok, sorry for some reason I read that the other way round; missed the `!=`. Thanks.